src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -376,6 +376,18 @@ public function initialized($id)
     }
 
     /**
+     * Clears shared services from the container.
+     */
+    public function clear()
+    {
+        if (!empty($this->scopedServices)) {
+            throw new RuntimeException('Clearing the container is not allowed when a scope is active.');
+        }
+
+        $this->services = array();
```

Comment:
I do understand that, I think the problem is more about the method name. `reset()` could be better?